src/APIFunctions/Messaging.js

```diff
@@ -0,0 +1,36 @@
+import axios from 'axios';
+import { ApiResponse } from './ApiResponses';
+
+let GENERAL_API_URL = process.env.REACT_APP_GENERAL_API_URL
+  || 'http://localhost:8080/api';
+
+export async function sendMessage(id, apiKey, message) {
+  let status = new ApiResponse();
+  const roomId = id || 'general';
+  await axios
+    .post(GENERAL_API_URL + '/messages/send', { apiKey, message, id: roomId })
+    .then(res => {
+      status.responseData = res.data;
+    })
+    .catch(err => {
+      status.error = true;
+      status.responseData = err;
+    });
+  return status;
+}
+
+export async function connectToRoom(room, apiKey, onMessage, onError) {
+  const eventSource = new EventSource(`${GENERAL_API_URL}/messages/listen?id=${encodeURIComponent(room)}&apiKey=${encodeURIComponent(apiKey)}`);
+
+  eventSource.onmessage = (event) => {
+    const data = JSON.parse(event.data);
```

Comment:
add a try/catch in here? or will `eventSource.onerror` handle paring errors for us?

Reply:
i think eventSource.onerror should handle the errors